iiblib/iibclient.py

```diff
@@ -303,20 +303,24 @@ def make_auth(self, iib_session):
 class IIBSession(object):
     """Helper class to support iib requests and authentication"""
 
-    def __init__(self, hostname, retries=3, verify=True):
+    def __init__(self, hostname, retries=3, verify=True, backoff_factor=2):
         """
         Args:
             hostname (str)
                 hostname of IIB service
             retries (int)
-                Number of http retries
+                number of http retries
+            verify (bool)
+                enable/disable SSL verification
+            backoff_factor (int)
+                backoff factor to apply between attempts after the second try
         """
         self.session = requests.Session()
         self.hostname = hostname
         self.verify = verify
 
         retry = Retry(
-            total=retries, read=retries, connect=retries, status_forcelist=[500]
```

Comment:
Does it make sense to retry for 4xx errors?

Reply:
Sorry I wrongly read the docs. I've added retry on 5xx status responses (500-511).